core/src/mender-artifact.c

```diff
@@ -1342,6 +1318,35 @@ artifact_drop_file(mender_artifact_ctx_t *ctx) {
     return MENDER_DONE;
 }
 
+static mender_err_t
+artifact_shift_and_checksum_data(mender_artifact_ctx_t *ctx, size_t length, const char *checksum_key, size_t checksum_len) {
+    assert(NULL != ctx);
+    assert(ctx->input.length >= length);
+    assert(checksum_len <= length);
+
+    if (0 == length) {
+        return MENDER_OK;
+    }
+
+#ifdef CONFIG_MENDER_FULL_PARSE_ARTIFACT
+    if ((NULL != checksum_key) && (0 != checksum_len)) {
+        mender_artifact_checksum_t *checksum;
+        /* Get checksum entry (create one if needed) */
+        if (NULL == (checksum = artifact_checksum_get_or_create(ctx, checksum_key))) {
+            /* Error already logged */
+            return MENDER_FAIL;
+        }
+
+        if (MENDER_OK != mender_sha256_update(checksum->context, ctx->input.data, checksum_len)) {
+            mender_log_error("Failed to update update checksum");
+            return MENDER_FAIL;
+        }
+    }
+#endif /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
```

Comment:
Not sure if these would trigger unused argument warnings if compiled without `CONFIG_MENDER_FULL_PARSE_ARTIFACT`
```suggestion
    }
#else /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
    (void)checksum_key;
    (void)checksum_len;
#endif /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
```

Reply:
I guess they would, good catch!